Sources/Build/BuildPlan.swift

```diff
@@ -1748,21 +1748,51 @@ public class BuildPlan {
         if let flags = pkgConfigCache[target] {
             return flags
         }
-        // Otherwise, get the result and cache it.
-        guard let result = pkgConfigArgs(for: target, diagnostics: diagnostics) else {
+        else {
             pkgConfigCache[target] = ([], [])
-            return pkgConfigCache[target]!
-        }
-        // If there is no pc file on system and we have an available provider, emit a warning.
-        if let provider = result.provider, result.couldNotFindConfigFile {
-            diagnostics.emit(.pkgConfigHint(pkgConfigName: result.pkgConfigName, installText: provider.installText))
-        } else if let error = result.error {
-            diagnostics.emit(
-                .warning(PkgConfigGenericDiagnostic(error: "\(error)")),
-                location: PkgConfigDiagnosticLocation(pcFile: result.pkgConfigName, target: target.name)
-            )
         }
-        pkgConfigCache[target] = (result.cFlags, result.libs)
+        let results = pkgConfigArgs(for: target, diagnostics: diagnostics)
+        var ret: [(cFlags: [String], libs: [String])] = []
+        for resultW in results {
+            // Otherwise, get the result and cache it.
+            guard let result = resultW else {
+                ret.append(([], []))
+                continue
+            }
+
+            // If there is no pc file on system and we have an available provider, emit a warning.
+            if let provider = result.provider, result.couldNotFindConfigFile {
+                diagnostics.emit(.pkgConfigHint(pkgConfigName: result.pkgConfigName, installText: provider.installText))
+            } else if let error = result.error {
+                diagnostics.emit(
+                        .warning(PkgConfigGenericDiagnostic(error: "\(error)")),
+                        location: PkgConfigDiagnosticLocation(pcFile: result.pkgConfigName, target: target.name)
+                )
+            }
+
+            ret.append((result.cFlags, result.libs))
+        }
+
+        // Build cache
+        var cflagsCache: [String] = []
+        var libsCache: [String] = []
+        for tuple in ret {
+            // Avoid duplicates while merging
```

Comment:
Would this work correctly if one library provides, for example, `-I foo` and the other provides `-I bar`?  In that case I think this code would end up with `-I foo bar`, which wouldn't work.  I'm not sure that the flags actually should be uniqued here, but I am not that familiar with what kind of flags are usually provided via PkgConfig or whether not uniquing them would cause problems.

Reply:
Yes here I removed duplicates for debugging. It is an issue.

Reply:
Understood.  With the latest changes, I think it's just the comment that needs to be removed then.  The looks could probably also be simplified to append the contents of the whole collection as well.